File: wrappers/mvc/src/KendoUI.Mvc.Web/UI/Upload/UploadLocalization.cs
namespace Telerik.Web.Mvc.UI
{
    using System.Globalization;
    using Telerik.Web.Mvc.Infrastructure;

    /// <summary>
    /// Localization strings for the Upload component
    /// </summary>
    public class UploadLocalization : ViewComponentLocalization, IClientSerializable
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="UploadLocalization" /> class.
        /// </summary>
        /// <param name="localizationService">The localization service.</param>
        /// <param name="culture">The culture.</param>
        public UploadLocalization(ILocalizationService localizationService, CultureInfo culture)
        : base(localizationService, null, "UploadLocalization", culture)
        {
        }

        /// <summary>
        /// Gets the Select string.
        /// </summary>
        /// <value>The default value is "Select...".</value>
        public string Select
        {
            get { return GetValue("Select"); }
        }

        /// <summary>
        /// Gets the Cancel string.
        /// </summary>
        /// <value>The default value is "Cancel".</value>
        public string Cancel
        {
            get { return GetValue("Cancel"); }
        }

        /// <summary>
        /// Gets the Retry string.
        /// </summary>
        /// <value>The default value is "Retry".</value>
        public string Retry
        {
            get { return GetValue("Retry"); }
        }

        /// <summary>
        /// Gets the Remove string.
        /// </summary>
        /// <value>The default value is "Remove".</value>
        public string Remove
        {
            get { return GetValue("Remove"); }
        }

        /// <summary>
        /// Gets the UploadSelectedFiles string.
        /// </summary>
        /// <value>The default value is "Upload files".</value>
        public string UploadSelectedFiles
        {
            get { return GetValue("UploadSelectedFiles"); }
        }

        /// <summary>
        /// Gets the DropFilesHere string.
        /// </summary>
        /// <value>The default value is "drop files here to upload".</value>
        public string DropFilesHere
        {
            get { return GetValue("DropFilesHere"); }
        }

        /// <summary>
        /// Gets the "uploading" status string.
        /// </summary>
        /// <value>The default value is "uploading".</value>
        public string StatusUploading
        {
            get { return GetValue("StatusUploading"); }
        }

        /// <summary>
        /// Gets the "uploaded" status string.
        /// </summary>
        /// <value>The default value is "uploaded".</value>
        public string StatusUploaded
        {
            get { return GetValue("StatusUploaded"); }
        }

        /// <summary>
        /// Gets the "failed" status string.
        /// </summary>
        /// <value>The default value is "failed".</value>
        public string StatusFailed
        {
            get { return GetValue("StatusFailed"); }
        }

        /// <summary>
        /// Serializes the localization strings.
        /// </summary>
        /// <param name="key">The key.</param>
        /// <param name="writer">The writer.</param>
        public void SerializeTo(string key, IClientSideObjectWriter writer)
        {
            if (!IsDefault)
            {
                writer.AppendObject(key, ToJson());
            }
        }
    }
}

File: wrappers/php/lib/Kendo/UI/UploadLocalization.php
<?php

namespace Kendo\UI;

class UploadLocalization extends \Kendo\SerializableObject {
//>> Properties

    /**
    * Sets the text of the cancel button text.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function cancel($value) {
        return $this->setProperty('cancel', $value);
    }

    /**
    * Sets the drop zone hint.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function dropFilesHere($value) {
        return $this->setProperty('dropFilesHere', $value);
    }

    /**
    * Sets the header status message for uploaded files.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function headerStatusUploaded($value) {
        return $this->setProperty('headerStatusUploaded', $value);
    }

    /**
    * Sets the header status message for files that are being uploaded.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function headerStatusUploading($value) {
        return $this->setProperty('headerStatusUploading', $value);
    }

    /**
    * Sets the text of the remove button text.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function remove($value) {
        return $this->setProperty('remove', $value);
    }

    /**
    * Sets the text of the retry button text.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function retry($value) {
        return $this->setProperty('retry', $value);
    }

    /**
    * Sets the "Select..." button text.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function select($value) {
        return $this->setProperty('select', $value);
    }

    /**
    * Sets the status message for failed uploads.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function statusFailed($value) {
        return $this->setProperty('statusFailed', $value);
    }

    /**
    * Sets the status message for uploaded files.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function statusUploaded($value) {
        return $this->setProperty('statusUploaded', $value);
    }

    /**
    * Sets the status message for files that are being uploaded.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function statusUploading($value) {
        return $this->setProperty('statusUploading', $value);
    }

    /**
    * Sets the text of the "Upload files" button.
    * @param string $value
    * @return \Kendo\UI\UploadLocalization
    */
    public function uploadSelectedFiles($value) {
        return $this->setProperty('uploadSelectedFiles', $value);
    }

//<< Properties
}

?>
